Elliot245/issues 89/add mcp protocol

elliot245 opened the pull request:
Fix https://github.com/nanobrowser/nanobrowser/issues/89

alexchenzl commented:
@elliot245 thank you for your awesome new feature!

The current repo is customized for building a Chrome extension. In order to make code clean and easy to maintain, I created a new repo https://github.com/nanobrowser/nanobrowser-mcp-host. 

1. Could you move the source code of the mcp-host package to the new repo https://github.com/nanobrowser/nanobrowser-mcp-host, and only keep extension related code in the current nanobrowser repo?
2. Also move doc folder to https://github.com/nanobrowser/nanobrowser-mcp-host
3. If possible, could you also provide an English version of those document in doc ?
4. Looks like the Popup window is conflict with the SidePanel. We need to keep SidePanel because most users use it and only advanced users will use MCP. So I suggest to implement MCP configuration in options and display status in SidePanel

elliot245 commented:
@alexchenzl I have split the mcp-host part of the code to https://github.com/nanobrowser/nanobrowser-mcp-host/pull/2, and moved the functions in the popup to options and side-panel. Please review it again.

elliot245 committed:
feat: clean up unused code

elliot245 committed:
feat: add mcp host installation information

elliot245 commented:
I squash commits and then add boot instructions for installing mcp-host.

elliot245 committed:
feat: format background index

elliot245 committed:
fix: fix open mcp settings not switch tab

alexchenzl commented:
> I squash commits and then add boot instructions for installing mcp-host.

Thank you! I will review them late this week.

elliot245 committed:
fix: fixbug for run_task no results

dbhurley commented:
Are we providing mcp client support within the chrome extension? That's desirable.

alexchenzl reviewed (commented):
1. About the MCP icon displayed on Sidepanel
 a. Add a configuration: "Display MCP Icon on Sidepanel", let users decide whether to display it because only advanced users will use it.
 b. then users can click this icon to start/stop MCP host
 c. it would be better to use different colors for connected/disconnected statuses for the icon
2. make text left-aligned in the configuration page
3. There are some eslint errors about imports, for example:
   import { McpError, McpErrorCode } from '../errors/mcp-errors';
   lint error: Imports "McpError" are only used as type

elliot245 committed:
feat: update mcp host connect app name

alexchenzl commented:
@elliot245 I've been giving more thought to the communication mechanism between the Nanobrowser extension and the MCP host, and I'm leaning towards WebSocket as potentially a more advantageous approach compared to Native Messaging.

**Native Messaging Concerns**:

- Complex User Setup: Requires system-specific registration beyond just running the host (especially tricky on Windows).
- Local Machine Only: MCP host must be on the same machine.

**WebSocket Advantages**:

- Simpler User Setup: Users just run the MCP host; no extra registration.
- Flexible Deployment: MCP host can run remotely (e.g., different machine, cloud)

Considering user experience and flexibility, WebSocket seems to offer significant benefits. The [nanomachine](https://github.com/reindent/nanomachine) project by @reindent is a good reference for this approach.

I'd suggest we should consider using WebSockets. What are your thoughts?

elliot245 committed:
feat: update mcp settings

elliot245 committed:
opt: optimize mcp icon

elliot245 committed:
fix: fix lint error

elliot245 committed:
feat: rebase from master

elliot245 committed:
feat: fix build error

elliot245 commented:
> 1. About the MCP icon displayed on Sidepanel
>    a. Add a configuration: "Display MCP Icon on Sidepanel", let users decide whether to display it because only advanced users will use it.
>    b. then users can click this icon to start/stop MCP host
>    c. it would be better to use different colors for connected/disconnected statuses for the icon
> 2. make text left-aligned in the configuration page
> 3. There are some eslint errors about imports, for example:
>    import { McpError, McpErrorCode } from '../errors/mcp-errors';
>    lint error: Imports "McpError" are only used as type

These issues have been fixed, please review it again.

alexchenzl reviewed (commented):
I tried to Start MCP Host but failed the following are steps:

1. I run the install.sh in the nanobrowser-mcp-host repo, on MacOS
2. Click Start MCP Host in MCP settings, it displays connected
3. In fact the mcp-host is not running by checking the listenting ports
4. As a result, I can not stop it
5. Restart the Chrome
6. Click Start MCP Host again
7. It displays "MCP Host Installation Required"

Potential issues:

1. Looks like the connection status was not checked correctly
2. There are issues in the installation script in the mcp host repo.

elliot245 committed:
fix: fix false connection status

elliot245 committed:
feat: add timeout for conn check

elliot245 committed:
feat: update mcp host SSE port to 9666

ivanfernadezm99 commented:
This feature is exactly what I need to connect Nanobrowser with my own bots and automations (like Telegram or WhatsApp). Please consider merging it soon! Thank you for pushing the project forward.

LousyBook94 commented:
ooo, is this gonna be done soon

michalkomar commented:
Closing this PR as it has been open since May 2025 with unresolved conflicts and review feedback. The MCP protocol feature needs to be re-implemented with current codebase structure after our recent architectural changes. Please create a new PR if this feature is still needed.

shafqatevo commented:
Is MCP server capability available now?
